conda/common/configuration.py

```diff
@@ -206,7 +210,7 @@ def __repr__(self):
         return str(vars(self))
 
     @abstractmethod
-    def value(self, parameter_obj):
+    def value(self, parameter_obj=None):
```

Comment:
This was to keep my type checker in PyCharm from showing an error. I can switch it back to keep it consistent.

Reply:
I'm not sure I follow PyCharm's reason to require a default for this case TBH

Reply:
@jezdez, I actually went back and looked at this more today. It was more than just a type checker thing. It has to do with the way that `EnvRawParameter` is handled differently than everything else.

`EnvRawParameter`'s implementation of `value` is the only one which actually does anything with the `parameter_obj` value.  Presumably, it does this to handle the "splitting" of environment variables with a delimiter. Interestingly enough, passing in `None` here doesn't break any of that.

To be extra certain, I also created a new test with a custom delimiter, and still everything works. So, I'm not exactly certain what is going on. You can checkout this method for yourself here:

https://github.com/conda/conda/blob/c9aa8ff6e365692e55961fa7477f06c3f0fc5178/conda/common/configuration.py#L230

I have since rewritten the method, `get_element_type_from_value` to explicitly pass in `None` and removed the default values of `None` for the other `value` definitions.